code_generator/End2End/GeneratedDependencies/Generated/improbable/testschema/ExhaustiveBlittableSingularTranslation.cs

```diff
@@ -168,7 +168,136 @@ public void OnComponentUpdate(ComponentUpdateOp<global::Improbable.TestSchema.Ex
                 }
 
                 componentData.DirtyBit = false;
-                view.UpdateComponent(entity, componentData, UpdatesPool);
+
+                view.SetComponentData(entity, componentData);
+
+                var componentFieldsUpdated = false;
+                if (update.field1.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field2.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field4.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field5.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field6.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field8.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field9.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field10.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field11.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field12.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field13.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field14.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field15.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field16.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+                if (update.field17.HasValue)
+                {
+                    componentFieldsUpdated = true;
+                }
+
+                if (componentFieldsUpdated)
+                {
+                    var gdkUpdate = new SpatialOSExhaustiveBlittableSingular.Update();
```

Comment:
This is probably fine for now, but we should look at optimising this in the future given that we have already looped over this. Bitmask maybe?

Reply:
Combined the two code blocks. We now preemptively make an update struct without knowing whether there will be field updates but that's probably fine.

Reply:
That seems good to me. An empty update is still an update.